velox/functions/lib/Re2Functions.h

```diff
@@ -39,11 +39,23 @@ enum class PatternKind {
   kPrefix,
   /// Fixed pattern preceded by one or more '%', such as '%foo', '%%%hello'.
   kSuffix,
+  /// Patterns matching '%{c0}%', such as '%foo%%', '%%%hello%'.
+  kFixedMiddle,
   /// Patterns which do not fit any of the above types, such as 'hello_world',
   /// '_presto%'.
   kGeneric,
 };
 
+struct PatternMetadata {
+  PatternKind patternKind;
+  // Contains the length of the fixed pattern for patterns of kind kFixed,
+  // kPrefix, and kSuffix. Contains the count of wildcard character '_' for
+  // patterns of kind kExactlyN and kAtLeastN. Contains 0 otherwise.
+  vector_size_t length;
+  // Contains the fixed pattern in patterns of kind kOptimizedPattern.
+  StringView fixedPattern = StringView();
```

Comment:
StringView is just a view over a valid string from somewhere. It requires the actual string to be in scope for use. What is the string backing this StringView ? It might be worth just having std::string here so that its use is not tied to anything else.

Reply:
Thanks for the suggestion, I used the default constructor for StringView here just so the parameter is initialized for patterns other than kFixedCenter. Only for patterns of type kFixedCenter, the StringView is backed by the fixed string in the pattern. Modified to use std::string.